src/email-only-signup.js

```diff
@@ -1,132 +1,284 @@
+		_forEach(o.discreetContentFocusables, el => {
+			toggleTabIndex(el, showCollapsed);
+		});
+
+		updateComponentAriaControls();
+	}
+
+	// toggle the inactive styles on dropdown component to allow mock 'placeholder'
+	function toggleComponentSelectInactive (event) {
+		let isPlaceholderSelected = (event.target.options[event.target.selectedIndex].getAttribute('placeholder') !== null);
+
+		if (event.type === 'focus') {
+			removeClass(SELECT_INACTIVE_CLASS, o.topicSelect);
 		}
 
-		function optionsFromData(el) {
-			const options = {};
-			Object.keys(defaultOptions).forEach(key => {
-				// convert optionKeyLikeThis to data-o-email-only-signup-option-key-like-this
-				const attr = 'data-o-email-only-signup-' + kebabCase(key);
-				if(el.hasAttribute(attr)) {
-					options[key] = el.getAttribute(attr);
-				}
-			});
-
-			return options;
+		if (event.type === 'blur' && isPlaceholderSelected) {
+			addClass(SELECT_INACTIVE_CLASS, o.topicSelect);
 		}
+	}
 
-		function serializeFormInputs(form) {
-			const inputs = form.elements;
-			let str = [];
+	/*
+	 * display a message in the component
+	 * @param {HTML String} message - content to inject
+	 */
+	function displayComponentMessage (message) {
+		o.displaySection.innerHTML = message;
+	}
 
-			for (let i=0; i<inputs.length; i++) {
-				let field = form.elements[i];
-				if (field.name && field.type !== 'submit' && field.type !== 'button') {
-					str.push(`${encodeComponent(field.name)}=${encodeComponent(field.value)}`);
-				}
+	//update 'aria-expanded' attr to match an 'aria-controls' element target state
+	function updateComponentAriaControls () {
+		_forEach(o.ariaControls, el => {
+			const target = o.self.querySelector('#' + el.getAttribute('aria-controls'));
+			if (target) {
+				const targetIsHidden = (target && target.getAttribute('aria-hidden') === 'true');
+				el.setAttribute('aria-expanded', !targetIsHidden);
 			}
+		});
+	}
+};
 
-			return str.join("&");
+/*
+ * Extract options from element data attributes
+ * @param {Element} el
+ * @param {Object} options - Containing keys to match attrs
+ * @returns {Object}
+ */
+function optionsFromData (el, opts) {
+	const options = {};
+	_forEach(Object.keys(opts), key => {
+		// convert optionKeyLikeThis to data-o-email-only-signup-option-key-like-this
+		const attr = 'data-o-email-only-signup-' + kebabCase(key);
+		if (el.hasAttribute(attr)) {
+			options[key] = el.getAttribute(attr);
 		}
+	});
+	return options;
+}
 
-		function toggleSelectInactive (event) {
-			let isPlaceholderSelected = (event.target.options[event.target.selectedIndex].getAttribute('placeholder') !== null);
+/*
+ * find and return an array of 'focusable' elements in a given element
+ * @param {Element} el
+ * @returns {Array} - Array of Elements, can be empty
+ */
+function findFoucsablesInEl (el) {
+	let arr = [];
+	_forEach(['input', 'button', 'a'], selector => {
+		let nodeList = el.querySelectorAll(selector);
+		if (nodeList && nodeList.length > 0) {
```

Comment:
True.
It is there as a defensive check against the `querySelector` not returning anything.
If removed, either the `toArray` function needs to become more robust in what it allows to be transformed. Or the following array concat needs to check the validity of the array that `toArray` returns. It seems simpler (seeing as this is the only case for it in the code at the moment) to instead to a check before using `toArray`.